src/compiler/types.ts

```diff
@@ -342,6 +342,9 @@ namespace ts {
         JSDocReturnTag,
         JSDocTypeTag,
         JSDocTemplateTag,
+        JSDocTypedefTag,
+        JSDocPropertyTag,
+        JSDocTypeLiteral,

```

Comment:
A `@typedef` is similar to a type alias in TS. In TS you can give an alias to an object type literal:
```ts
type T = { name: string, age: number}
```
which is equivalent to the following jsdoc:
```js
/** 
 * @typedef {Object} T
 * @property {string} name
 * @property {number} age
 */
```
So I defined the `@property`s as `JSDocTypeLiteral`, as it functions like object type literal, but with a different shape.